### ISSUE TITLE
(solved) Error react-admin v4 and Next.js setup - ReferenceError: window is not defined

### ISSUE THREAD

**Issue by AtAFork** [marmelab/react-admin#7349]
I'll keep this issue here for others to find since I struggled with it (through no fault of the react-admin team). I was't running into the issue with react-admin v3, but came across it when upgrading to v4.

**What you were expecting:** 
React Admin 4.0.0-beta.2.0 to run properly using Next.js

**What happened instead:**
`Uncaught at testLocalStorage`
`error - ReferenceError: window is not defined`

I saw this error even though I dynamically imported react-admin into my `_app.tsx`:
```
const ReactAdmin = dynamic(() => import('react-admin'), {
  ssr: false,
});
```

The error occurred in `ra-core`'s `/dist` here:
https://github.com/marmelab/react-admin/blob/9a5fe983b1b1b65eccde3055b013db7627bda9dc/packages/ra-core/src/store/localStorageStore.ts#L14

**SOLUTION:**
You must make sure that you're not also importing react-admin in your `index.tsx` in Next.js `/src/pages` folder (or if you are, that you're dynamically importing it there too).

Hope this helps someone else!
-Adam

**Comment by fzaninotto** [comment]
There is indeed a bug in react-admin, let me reopen it.

**Comment by WiXSL** [comment]
Fixed by #7359
